ui/app/pages/swaps/slippage-buttons/slippage-buttons.js

```diff
@@ -6,12 +6,25 @@ import ButtonGroup from '../../../components/ui/button-group';
 import Button from '../../../components/ui/button';
 import InfoTooltip from '../../../components/ui/info-tooltip';
 
-export default function SlippageButtons({ onSelect, maxAllowedSlippage }) {
+export default function SlippageButtons({
+  onSelect,
+  maxAllowedSlippage,
+  currentSlippage,
+}) {
   const t = useContext(I18nContext);
-  const [open, setOpen] = useState(false);
   const [customValue, setCustomValue] = useState('');
   const [enteringCustomValue, setEnteringCustomValue] = useState(false);
-  const [activeButtonIndex, setActiveButtonIndex] = useState(1);
+  const [activeButtonIndex, setActiveButtonIndex] = useState(() => {
+    if (currentSlippage === 3) {
+      return 1;
+    } else if (currentSlippage === 2) {
+      return 0;
+    } else if (currentSlippage) {
```

Comment:
I think it might be best not to call `setCustomValue` in the callback to `useState`, as this is incidental to the purpose of this function. I think the intent of the code would be clearer if we added a callback to the `useState` call that sets up the `customValue` which would set the initial custom value to `currentSlippage` if `currentSlippage` is truthy but not equal to 2 or 3.

Reply:
I like that!

Reply:
Done.